src/adagrams.js

```diff
@@ -1,15 +1,218 @@
+      V: 4,
+      W: 4,
+      X: 8,
+      Y: 4,
+      Z: 10,
+    };
+  }
+  
+  drawLetters() {
+    // Build list of all the letters
+    const allLetters = [];
+    for (const [letter, num] of Object.entries(this.letterPool)) {
+      for (let i = 0; i < num; ++i) {
+        allLetters.push(letter);
+      }
+    }
 
-export const highestScoreFrom = (words) => {
-  // Implement this method for wave 4
-};
+    // build the hand making sure at least 3 letters are vowels
+    let hand = [];
+    let vowelCount = 0;
+    while (vowelCount < 3) {
+      hand = [];
+      for (let i = 0; i < 10; ++i) {
+        let randomLetter =
+          allLetters[Math.floor(Math.random() * allLetters.length)];
+
+        // make sure we dont exceed the number allowed for that letter
+        let randomLetterCountInHand = hand.filter(
+          (x) => x === randomLetter
+        ).length;
+        while (randomLetterCountInHand === this.letterPool[randomLetter]) {
+          randomLetter =
+            allLetters[Math.floor(Math.random() * allLetters.length)];
+          randomLetterCountInHand = hand.filter((x) => x === randomLetter).length;
+        }
+        // add the letter to the hand
+        hand.push(randomLetter);
+      }
+      // makes sure vowel_count is updated before we go through the loop again.
+      vowelCount = 0;
+      for (let vowel of this.vowels) {
+        vowelCount = vowelCount + hand.filter((x) => x === vowel).length;
+      }
+    }
+    return hand;
+  };
+
+  usesAvailableLetters(input, lettersInHand) {
+    for (let letter of input.toUpperCase()) {
+      // Check if the letter is in the hand
+      if (!lettersInHand.includes(letter)) {
+        return false;
+      }
+
+      // Check if there are more of the letter in the input than the hand
+      let numLetterInput = input.split("").filter((x) => x === letter).length;
+      let numLetterInHand = lettersInHand.filter((x) => x === letter).length;
+      if (numLetterInput > numLetterInHand) {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  scoreWord(word) {
+    let score = 0;
+    // if (!word) {
+    //   return score;
+    // }
+    for (let letter of word.toUpperCase()) {
+      score += this.letterValues[letter];
+    }
+    if (word.length > 6) {
+      score += 8;
+    }
+
+    return score;
+  };
+
+  highestScoreFrom(words) {
+    let winningWord = { word: "", score: 0 };
+    for (let word of words) {
+      let newScore = this.scoreWord(word);
+
+      if (word.length === 10) {
+        winningWord.word = word;
+        winningWord.score = newScore;
+        return winningWord;
+      }
+
+      if (newScore > winningWord.score) {
+        winningWord.word = word;
+        winningWord.score = newScore;
+      } else if (newScore === winningWord.score) {
+        if (word.length < winningWord.word.length) {
+          winningWord.word = word;
+          winningWord.score = newScore;
+        }
+      }
+    }
+
+    return winningWord;
```

Comment:
Great work on translating your python adagrams to JS. As you probably notice, I didn't leave many comments because the main purpose of this project really to just translate the code from one language to another. With that being said, if you have any questions about anything please feel free to reach out! (Great work on you making it into a class as well). ⭐️